### ISSUE TITLE
Nearby can not track current location or it takes too long to do

### ISSUE THREAD

**Issue by neslihanturan** [commons-app/apps-android-commons#4465]
**Summary:** 
This issue is created based on a feedback in our mailing list. They say usually the Commons App fails to follow them. They must flick the app out of memory and restart it, for it to know their location. It also takes much longer than any other app in their phone ( Strava, Google Maps, MSN Weather) to find their location. Thus they still must use WikiShootMe despite its disadvantages. 
Most often they use Commons App to snap and upload, after using WSM to find a target in whatever place they are visiting. 

So the issue is making sure current location is followed by the app and checking if there are any improvements that we can made to make it quicker.

**Comment by nicolas-raoul** [comment]
> _fails to follow them_

As a heavy user of Nearby, I agree that the blue dot does not follow me correctly. I very often have to close the app and restart it in order to see the blue dot at my current location, even though other apps like OsmAnd show my correct location in real time.

> _takes much longer than any other app in their phone ( Strava, Google Maps, MSN Weather) to find their location_

What takes a lot of time might be the SPARQL request? Are we showing the user's position in parallel with the SPARQL query? It seems to me that the app is stuck on the screen below for a long time, but I am not sure what it is waiting for (GPS? SPARQL?):
![adb](https://user-images.githubusercontent.com/99590/122629242-8d888a00-d0f6-11eb-974a-a153cb03c0c2.png)

**Comment by Rishavgupta12345** [comment]
nearby works perfectly now. It's an old issue and may be solved by any other issue. Please review and close this issue @nicolas-raoul

**Comment by nicolas-raoul** [comment]
@Rishavgupta12345 Have you read the issue carefully? This is about the current location tracking, which is still laggy in my experience.

**Comment by savsch** [comment]
A lot of changes have been made to nearby over the past year, and the blue-dot updation logic appears to have long lost sync with these changes.

**0.** The blue dot must match the user's current location to the best of the application's knowledge (GPS), irrespective of any ongoing query.

It is relatively straightforward to make the necessary fixes for point **0** to hold. However, for what it's worth, I believe there should be further enhancements to it:

The map should have two modes (**1** and **2**):
**1.** _Follow user's live location_: In this mode, the map adjusts its own position so as to keep the blue dot _at or near_ the center of the map.
**2.** _Free scrolling mode_: The behavior currently shown by the app.

- Pressing the recenter FAB should be the (only) way of entering mode 1, and it should also be the default mode when nearby first loads (if location is available).
- Any scrolling/resizing action by the user should lead to the map entering mode 2.
- Mode 1 can be indicated by a blue tint on the recenter FAB.
- I said "near" in '_at or near_' above because slight deviations from the center should not make the map move, to minimize unnecessary queries. 

Also, when the location is not available, the app is always in mode 2..

---

Realized I typed too much, so this might merit a separate issue for further discussion, if the suggestions are at all on-topic.

**Comment by savsch** [comment]
Adding to my last comment:
That way, #5734 can also be taken care of.
@nicolas-raoul I'd love to hear your thoughts when you get a chance!

**Comment by nicolas-raoul** [comment]
@savsch Sorry for the delay! All of what you wrote is totally on point! :-)

**Comment by sonalyadav1** [comment]
Will work on this issue.

**Comment by sonalyadav1** [comment]
Hi @nicolas-raoul !
 I noticed that #6312  appears to address the same problem, with an existing pull request already open. Could you kindly confirm whether these are indeed the same issue, or if there’s a distinction I’m missing?

**Comment by nicolas-raoul** [comment]
You are right!
Would you mind taking a few hours to test the change (you will need to walk around while looking at the blue dot) and write a code review? 🙂

**Comment by sonalyadav1** [comment]
Of course, I’ll do that . Thanks ! :)

**Comment by nicolas-raoul** [comment]
@sonalyadav1 Did you get a chance to perform the above? Please post your review as a comment in the pull request. Thanks a lot! 🙂

**Comment by nicolas-raoul** [comment]
Let's mark this as fixed and track remaining bugs in separate tickets. :-)
